rdtools/soiling.py

```diff
@@ -1052,3 +1065,1532 @@ def monthly_soiling_rates(soiling_interval_summary, min_interval_length=14,
+        |                        | seasonal component (SC)                      |
+        +------------------------+----------------------------------------------+
+        | 'seasonal_high'        | upper bound of 95 % conf. interval of        |
+        |                        | seasonal component (SC)                      |
+        +------------------------+----------------------------------------------+
+        | 'model_high'           | upper bound of 95 % confidence interval of   |
+        |                        | the model fit                                |
+        +------------------------+----------------------------------------------+
+        | 'model_low'            | lower bound of 95 % confidence interval of   |
+        |                        | the model fit                                |
+        +------------------------+----------------------------------------------+
+
+    degradation : list
+        List of linear degradation rate of system in %/year, lower and
+        upper bound of 95% confidence interval
+    soiling_loss : list
+        List of average soiling losses over the time series in %, lower and
+        upper bound of 95% confidence interval
+    residual_shift : float
+        Mean value of residuals. Multiply total model by this number for
+        complete overlap with input pi
+    RMSE : float
+        Root Means Squared Error of total model vs input pi
+    small_soiling_signal : bool
+        Whether or not the signal is deemed too small to infer anything
+        about it
+    adf_res : list
+        The results of an Augmented Dickey-Fuller test (telling whether the
+        residuals are stationary or not)
+    knobs_n_weights : pandas.DataFrame
+        Contains information about the knobs used in each bootstrap model
+        fit, and the resultant weight
+
+    References
+    ----------
+    .. [1] Skomedal, Å. and Deceglie, M. G., IEEE Journal of
+       Photovoltaics, Sept. 2020. https://doi.org/10.1109/JPHOTOV.2020.3018219
+    '''
+
+    def __init__(self, energy_normalized_daily):
+        self.pm = energy_normalized_daily  # daily performance metric
+
+        if np.isnan(self.pm.iloc[0]):
+            first_keeper = self.pm.isna().idxmin()
+            self.pm = self.pm.loc[first_keeper:]
+
+        if self.pm.index.freq != 'D':
+            raise ValueError('Daily performance metric series must have '
+                             'daily frequency (missing dates should be '
+                             'represented by NaNs)')
+
+    def iterative_signal_decomposition(
+            self, order=('SR', 'SC', 'Rd'), degradation_method='YoY',
+            max_iterations=18, detection_tuner=.5, convergence_criterium=5e-3,
+            pruning_iterations=1, pruning_tuner=.6, soiling_significance_knob=.75,
+            process_noise=1e-4, renormalize_SR=None, ffill=True, clip_soiling=True,
+            verbose=False):
+        '''
+        Description
+        -----------
```

Comment:
Let's remove this header, it doesn't get displayed in the rendered sphinx output and it messes up the single-line description of this function